central/notifiers/jira/jira.go

```diff
@@ -320,53 +493,96 @@ func newJira(notifier *storage.Notifier, metadataGetter notifiers.MetadataGetter
 			}
-			prios, _, err = client.Priority.GetList()
+
+			req, err = client.NewRequest("GET", urlPath, nil)
+			if err != nil {
+				return nil, err
+			}
+			_, err = client.Do(req, nil)
 		}
 		if err != nil {
-			return nil, errors.Wrap(err, "could not get the priority list")
+			if strings.HasPrefix(errStr, "401") || strings.HasPrefix(errStr, "403") {
+				return nil, errors.Wrap(err, "Could not authenticate to Jira")
+			}
 		}
 	}
-	jiraConf := notifier.GetJira()
-
-	derivedPriorities := mapPriorities(jiraConf, prios)
-	if len(jiraConf.GetPriorityMappings()) == 0 {
-		bytes, _ := json.Marshal(&derivedPriorities)
-		log.Debugf("Derived Jira Priorities: %s", bytes)
-		for k, v := range derivedPriorities {
-			jiraConf.PriorityMappings = append(jiraConf.PriorityMappings, &storage.Jira_PriorityMapping{
-				Severity:     k,
-				PriorityName: v,
-			})
+
+	return client, nil
+}
+
+func canCreateIssuesInProject(client *jiraLib.Client, project string) (bool, error) {
+	urlPath := fmt.Sprintf("rest/api/2/mypermissions/?projectKey=%s&permissions=CREATE_ISSUES", project)
+
+	req, err := client.NewRequest("GET", urlPath, nil)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debugf("Making request to %s", urlPath)
+	resp, err := client.Do(req, nil)
+	if err != nil {
+		log.Debugf("Raw error message from jira lib: %s", err.Error())
+		if resp != nil && resp.StatusCode == 404 {
+			return false, fmt.Errorf("Project %s not found", project)
 		}
-		sort.Slice(jiraConf.PriorityMappings, func(i, j int) bool {
-			return jiraConf.PriorityMappings[i].Severity < jiraConf.PriorityMappings[j].Severity
-		})
+		return false, err
 	}
 
-	needsPriority, err := isPriorityNeeded(client, notifier.GetLabelDefault(), jiraConf.GetIssueType())
+	result := &permissionResult{}
+
+	defer utils.IgnoreError(resp.Body.Close)
+	err = json.NewDecoder(resp.Body).Decode(result)
 	if err != nil {
-		return nil, errors.Wrapf(err, "could not determine if priority is a required field for project %q issue type %q", notifier.GetLabelDefault(), jiraConf.GetIssueType())
+		return false, err
 	}
 
-	// marshal unknowns
-	var unknownMap map[string]interface{}
-	if jiraConf.GetDefaultFieldsJson() != "" {
-		if err := json.Unmarshal([]byte(jiraConf.GetDefaultFieldsJson()), &unknownMap); err != nil {
-			return nil, errors.Wrap(err, "could not unmarshal default fields JSON")
+	return result.Permissions["CREATE_ISSUES"].HavePermission, nil
+}
+
+func configurePriority(client *jiraLib.Client, jiraConf *storage.Jira, project string) (map[storage.Severity]string, error) {
+	hasPriority, err := isPriorityFieldOnIssueType(client, project, jiraConf.GetIssueType())
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not determine if priority is a required field for project %q issue type %q", project, jiraConf.GetIssueType())
+	}
+
+	if !hasPriority {
+		errMsg := "Priority field not found on requested issue type %s in project %s. Consider checking the 'Disable setting priority' box."
```

Comment:
Any reason to have an intermediate var here? Line too long otherwise?

Reply:
yeah just line length.  without the intermediate var it's the longest line at 188 chars.